### PR TITLE
#886 Avoid usage of null value.

### PR DESCRIPTION
**Pull request by VsSekorin**
As per #886

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#918](https://github.com/yegor256/cactoos/issues/918) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/1841639326); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/org/cactoos/func/RetryProcTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * @since 0.8
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 line)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public final class RetryProcTest {
+
+    @Test
+    public void runsProcDefaultMultipleTimes() throws Exception {
```
Comment (victornoel) [inline src/test/java/org/cactoos/func/RetryProcTest.java]: @VsSekorin please add more tests as originally.
If you don't want to rely on some non-deterministic behaviour, then you can test it fails after a fixed number of times for example.

File: src/test/java/org/cactoos/func/RetryProcTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        final AtomicInteger number = new AtomicInteger(0);
+        new RetryProc<>(AtomicInteger::incrementAndGet).exec(number);
+        MatcherAssert.assertThat(
+            number.get(),
+            Matchers.equalTo(1)
```
Comment (victornoel) [inline src/test/java/org/cactoos/func/RetryProcTest.java]: @VsSekorin prefer to use an OO version of this matcher (`IsEqual` class)

**Comment by llorllale** [comment]
@victornoel I don't understand what you wrote

**Comment by victornoel** [comment]
@llorllale sorry about that, I will try again.

In this PR, if you look at the code of `RetryProc`, you can see it is almost exactly the same as the existing code of `RetryFunc`.
It makes sense, because they have exactly the same behaviour: retrying when there are errors.

I'm saying I am bothered by this situation and that I don't think duplication is a good solution just for the sake of removing `null` in a language like Java, but I'm not sure how to do things differently.
I suppose you could write `RetryProc` based on `RetryFunc`, with a return type of `Void`, hence my rant about `Void`. But Java's `Void` has only one acceptable value: `null`.

Other issues have the same question (`TimedFunc`/`TimedProc`, #887 for example).

Is this clearer?

**Comment by llorllale** [comment]
@victornoel I agree with your assessment. It aligns rather well with #915.

Let's just simply drop these ctors that force `null` and forget about `RetryProc`. Following Yegor's principle of 0%/100%: we either accept `null` or we reject `null` everywhere. And we choose the latter.

Users will have to work around this with a little bit more work by using `RetryFunc`:

```java
new RetryFunc(
    { input -> /* do something */; return true; }
)
```

Same guidelines apply for #887.

**Comment by victornoel** [comment]
@VsSekorin so to conclude this review, please take into account the feedbacks from @llorllale about dropping `RetryProc`.

**Comment by victornoel** [comment]
@VsSekorin ping

**Comment by victornoel** [comment]
@VsSekorin ping

**Comment by victornoel** [comment]
@VsSekorin ping

**Comment by llorllale** [comment]
@victornoel 

@VsSekorin was resigned from this job, so I'm closing this PR

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#918](https://github.com/yegor256/cactoos/issues/918) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by elenavolokhova** [comment]
@0crat quality acceptable

**Pull request closed without merging**